File: csharp/Lagerverwaltung/WerhausCore/Warehouse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace WerhausCore
{
    public class Warehouse
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int Location { get; set; }
        public int Capacity { get; set; }
        public Owner Owner { get; set; }
        public List<Product> Products { get; set; }
        public List<Order> Orders { get; set; }
        private int currentCapacity = 0;
        public int CurrentCapacity
        {
            get
            {
                return currentCapacity;
            }
            set
            {
                currentCapacity = value;
            }
        }

        public Warehouse(string name, string description, int location, int capacity, Owner owner, List<Product> products, List<Order> orders)
        {
            Name = name;
            Description = description;
            Location = location;
            Capacity = capacity;
            Owner = owner;
            Products = products;
            Orders = orders;
            currentCapacity = GetCurrentCapacity();
        }
        //public Warehouse(string name, string description, int location, int capacity, Owner owner)
        //{
        //    Name = name;
        //    Description = description;
        //    Location = location;
        //    Capacity = capacity;
        //    Owner = owner;
        //    Products = new List<Product>();
        //    Orders = new List<Order>();
        //    currentCapacity = GetCurrentCapacity();
        //}
        public int GetCurrentCapacity()
        {
            if (Products != null)
            {
                int sum = 0;

                foreach (Product p in Products)
                {
                    sum += p.Space * p.Amount;
                }
                return sum;
            }
            return 0;
        }
        public override string ToString()
        {
            return Name + " with a capacity of " + Capacity + "\n" + Description;
        }
    }
}

File: webservice/routing-layer/warehouse.js
// packages //TODO: integrate into user.js and remove
const express = require('express');
const oracleConnection = require('../data-layer/oracleDataAccess');
const classParser = require('../data-layer/classParser');
const classes = require('../data-layer/classes');
const router = express.Router();

// add routes
router.route('/')
    .get((req, res) => {
        let query = 'SELECT * from SW_Warehouse',
            param = [];

        oracleConnection.execute(query, param,
            (result) => res.status(200).json(classParser(result.rows, classes.Warehouse)),
            (err) => res.status(404).json({
                message: err.message,
                details: err
            })
        );
    }).post((req, res) => {
        let query = 'INSERT INTO SW_Warehouse VALUES (seq_warehouse.NEXTVAL, :name, :descripion, :capacity, :id_owner)',
            param = [req.body.name, req.body.description, req.body.capacity, req.body.id_owner || req.body.owner.id];

        oracleConnection.execute(query, param,
            (result) => res.status(201).json(result),
            (err) => res.status(500).json({
                message: err.message,
                details: err
            }));
    });

router.route('/:id')
    .get((req, res) => {
        let query = 'SELECT * from SW_Warehouse WHERE id = :id',
            param = [req.params.id],
            innerQuery = 'SELECT id_product, name, description, price, space, amount from SW_Stored_In INNER JOIN SW_Product ON id_product = SW_Product.id WHERE id_warehouse = :id';

        oracleConnection.execute(query, param,
            (result) => {
                let warehouse = classParser(result.rows, classes.FatWarehouse)[0];

                if (!warehouse)
                    res.sendStatus(404);
                else
                    oracleConnection.execute(innerQuery, param,
                        (result) => {
                            warehouse.products = classParser(result.rows, classes.Product);

                            res.status(200).json(warehouse);
                        },
                        (err) => res.status(404).json({
                            message: err.message,
                            details: err
                        })
                    );
            },
            (err) => res.status(404).json({
                message: err.message,
                details: err
            })
        );
    });

module.exports = router;